plugin/src/function.py

```diff
@@ -0,0 +1,62 @@
+import requests
+import time
+
+def main(kwargs):
+
+    print("[INFO] Info recieved: {}".format(kwargs))
+    
+    if len(kwargs) < 4:
+        print("[ERROR] One or more parameters are missing")
+        return {"status": "error"}
+    
+    result = get_currency(**kwargs)
+
+    args = result.get("rates")
+
+    print (args)
+       
+    req = update_device(args, **kwargs)
+    del kwargs
+
+    return req.json()
+
+def get_currency(currencies, base, _plugin_env_API_URL, **kwargs):
+    
+    url = "{}/latest?base={}&symbols={}".format(_plugin_env_API_URL, base, currencies)
+    headers = {"Content-Type": "application/json"}
+    req = create_request(url, headers, attempts=5, request_type="get")
```

Comment:
if the server is down, the return will throw an exception. Add a try-except loop